### PR TITLE
chore: move input parameters to aurora-engine-types

### PR DESCRIPTION
**Pull request by aleksuss**
## Description

The goal of the PR is to allow to use input parameters of the engine without importing the `aurora-engine` itself. 

## Performance / NEAR gas cost considerations

There are no changes because it's refactoring only.

## Testing

Some tests were moved only. 

## How should this be reviewed

Pay attention that parameters correctly have been placed in corresponding modules.

### CODE REVIEW

**Review by mandreyel — APPROVED** [review]

File: engine-types/src/parameters/connector.rs [hunk 1]
```diff
@@ -0,0 +1,250 @@
+use base64::Engine;
+#[cfg(not(feature = "borsh-compat"))]
+use borsh::{BorshDeserialize, BorshSerialize};
+#[cfg(feature = "borsh-compat")]
+use borsh_compat::{self as borsh, BorshDeserialize, BorshSerialize};
+use primitive_types::{H160, H256};
```
Comment (mrLSD) [inline engine-types/src/parameters/connector.rs:6]: I think better:
```
use crate::{H160, H256};
```

File: engine-types/src/parameters/mod.rs [hunk 2]
```diff
@@ -1,330 +1,5 @@
-use crate::{
-    account_id::AccountId,
-    types::{Address, NEP141Wei, NearGas, RawU256, Yocto},
-    Box, String, Vec,
-};
-#[cfg(not(feature = "borsh-compat"))]
-use borsh::{maybestd::io, BorshDeserialize, BorshSerialize};
-#[cfg(feature = "borsh-compat")]
-use borsh_compat::{self as borsh, maybestd::io, BorshDeserialize, BorshSerialize};
+pub use promise::*;
```
Comment (mrLSD) [inline engine-types/src/parameters/mod.rs:1]: Why it's a wildcard?

Comment (mrLSD) [reply engine-types/src/parameters/mod.rs:1]: Maybe just:
```
pub use promise;
```
It will be more precise in importing.

Comment (aleksuss) [reply engine-types/src/parameters/mod.rs:1]: That's for saving backward compatibility.

**Review by mrLSD — APPROVED** [review]

**Pull request merged**